### ISSUE TITLE
JDK8 discussion

### ISSUE THREAD

**Issue by smadasu** [redis/jedis#1863]
Switching the version to 1.8 is trivial in pom.xml. What else is preventing us to do that?

**Comment by marcosnils** [comment]
I guess nothing. Wanna give it a shot.

**Comment by smadasu** [comment]
Sure.

**Comment by Martian101** [comment]
I encountered this error when using `public String set(final String key, final String value, final SetParams params)` method in JDK 1.8.0_111:
```java
java.lang.ClassCastException: java.lang.Integer cannot be cast to [C

	at com.gexin.clients.jedis.params.set.SetParams.getByteParams(SetParams.java:75)
	at com.gexin.clients.jedis.BinaryClient.set(BinaryClient.java:106)
	at com.gexin.clients.jedis.Client.set(Client.java:53)
	at com.gexin.clients.jedis.Jedis.set(Jedis.java:136)
```
It seems that a type inference error occured when the type parameter is used in the return type:
```java
  public <T> T getParam(String name) {
    if (params == null) return null;

    return (T) params.get(name);
  }
```
I can ensure this issue after some test both in JDK 1.7 and 1.8:
``` java
    @Test
    public void testT() {
        System.out.println(String.valueOf(getParam()));
    }

    public static <T> T getParam() {
        return (T) "AAA";
    }
```

**Comment by JensenZz** [comment]
@Martian101   maybe you can find your answer      https://stackoverflow.com/questions/23155907/classcast-error-java-7-vs-java-8

**Comment by sazzad16** [comment]
Switching to JDK1.8 will be required if we want to upgrade Commons Pool to version `2.7.0`.

References:
- #2039 
- #2040

**Comment by sazzad16** [comment]
Resolved by #2396
